packages/app/src/elements/app.ts

```diff
@@ -126,6 +129,11 @@ export class App extends ServiceWorker(StateMixin(AutoSync(ErrorHandling(AutoLoc
             return;
         }
 
+        if (page === "items") {
+            // Run audits, if necessary
+            this._maybeRunAudits();
+        }
```

Comment:
I don't like having this here, but maybe it's fine. I initially tried to do it in `core/src/app:_unlocked`, but that meant I'd have to import `@padloc/app` inside for `auditItems`, or refactor that to not need access to `app.save()` and `app.saveVault()` (or pass those functions inside, which would be doable, but ugly), and I didn't like that.

I also tried hooking into `stateChanged()` in this element, but that triggered more frequently.

I hope you have a better suggestion @MaKleSoft ?

Reply:
I'd also move `auditVaults` from `src/elements/audit.ts` into a method on a new mixin under `mixins/audit.ts` and add that mixin to the `<pl-app>` element. (The audit element doesn't need to have any of the actual audit logic but should only take care of displaying results).

Reply:
I hear you, that makes sense.